src/test/java/picard/sam/markduplicates/MarkDuplicatesTagRepresentativeReadIndexTest.java

```diff
@@ -36,69 +38,76 @@ protected MarkDuplicatesTagRepresentativeReadIndexTester getTester() {
-        tester.expectedRepresentativeIndexMap.put(3, representativeReadIndexInFileForward);
+        tester.expectedRepresentativeReadMap.put(representativeReadName1, representativeReadName1);
         tester.expectedSetSizeMap.put(representativeReadName1,2);
-        tester.addMatePair("RUNID:1:1:15993:13361", 1, 485253, 485253, false, false, true, true, "44M1S", "44M1S", false, true, false, false, false, DEFAULT_BASE_QUALITY);
-        tester.expectedRepresentativeIndexMap.put(0, representativeReadIndexInFileForward);
-        tester.expectedRepresentativeIndexMap.put(2, representativeReadIndexInFileForward);
-        tester.expectedSetSizeMap.put("RUNID:1:1:15993:13361",2);
+        final String duplicateSet1DuplicateReadName = "RUNID:1:1:15993:13361";
+        tester.addMatePair(duplicateSet1DuplicateReadName, 1, 485253, 485253, false, false, true, true, "44M1S", "44M1S", false, true, false, false, false, DEFAULT_BASE_QUALITY);
+        tester.expectedRepresentativeReadMap.put(duplicateSet1DuplicateReadName, representativeReadName1);
+        tester.expectedSetSizeMap.put(duplicateSet1DuplicateReadName,2);
 
         // Duplicate set: size 3 - all optical
-        String representativeReadName2 = "RUNID:1:1:15993:13360";
-        Integer representativeReadIndexInFileForward2 = 6;
+        final String representativeReadName2 = "RUNID:1:1:15993:13360";
         tester.addMatePair(representativeReadName2, 1, 485299, 485299, false, false, false, false, "45M", "45M", false, true, false, false, false, DEFAULT_BASE_QUALITY);
-        tester.expectedRepresentativeIndexMap.put(representativeReadIndexInFileForward2, representativeReadIndexInFileForward2);
-        tester.expectedRepresentativeIndexMap.put(9, representativeReadIndexInFileForward2);
+        tester.expectedRepresentativeReadMap.put(representativeReadName2, representativeReadName2);
         tester.expectedSetSizeMap.put(representativeReadName2,3);
-        tester.addMatePair("RUNID:1:1:15993:13365", 1, 485299, 485299, false, false, true, true, "44M1S", "44M1S", false, true, false, false, false, DEFAULT_BASE_QUALITY);
-        tester.expectedRepresentativeIndexMap.put(7, representativeReadIndexInFileForward2);
-        tester.expectedRepresentativeIndexMap.put(10, representativeReadIndexInFileForward2);
-        tester.expectedSetSizeMap.put("RUNID:1:1:15993:13365",3);
-        tester.addMatePair("RUNID:1:1:15993:13370", 1, 485299, 485299, false, false, true, true, "43M2S", "43M2S", false, true, false, false, false, DEFAULT_BASE_QUALITY);
-        tester.expectedRepresentativeIndexMap.put(8, representativeReadIndexInFileForward2);
-        tester.expectedRepresentativeIndexMap.put(11, representativeReadIndexInFileForward2);
-        tester.expectedSetSizeMap.put("RUNID:1:1:15993:13370",3);
+        final String duplicateSet2DuplicateReadName1 = "RUNID:1:1:15993:13365";
```

Comment:
bleh is this the best of how our testing for this feature worked before? this seems very brittle... not really your problem to fix here though...